### PR TITLE
Add a priority option to AutoArpa

### PR DESCRIPTION
**Pull request by wjgauthier**
I have made the changes we talked about in https://github.com/octodns/octodns/issues/1158. I hope this makes sense. Just two things:
- I had to change all the `assertEqual` statements in `test_octodns_processor_arpa.py` because `process_source_zone` now returns a priority
- Because `self._records` is a list now, it can contain duplicate values for the same IP, so I convert it to a set to ensure unique values (the test_dynamic test would fail because of duplicate values)

### CODE REVIEW

File: octodns/processor/arpa.py [hunk 1]
```diff
@@ -37,7 +39,10 @@ def process_source_zone(self, desired, sources):
 
                 for ip in ips:
                     ptr = ip_address(ip).reverse_pointer
-                    self._records[f'{ptr}.'].add(record.fqdn)
+                    auto_arpa_priority = record.octodns.get('auto_arpa_priority', 999)
+                    self._records[f'{ptr}.'].append((auto_arpa_priority, record.fqdn))
+                    unique_list = list(set(self._records[f'{ptr}.']))
```
Comment (ross) [inline octodns/processor/arpa.py:44]: I believe this will only keep things unique when they have the same priority since tuples of (priority, fqdn) are being placed into the list. so if the same fqdn showed up 

It may be preferable to just `append` everything here and then deal with making things unique down below when it's pulling the fqdn's out of the tuples. That'd probably result in a slightly more predictable behavior where the lowest priority seen for a fqdn would win.

Another option that comes to mind would be a `dict` with the fqdn a the key and priority as the value. That would result in last one winning though which is at least consistent, but probably harder to predict. Sorting it would require a couple of extra steps, something like `.items()`, flipping the k & v, and then sorting.

**Review by ross — COMMENTED** [review]
Mostly there. One comment/discussion bit inline. 

Thanks.

File: octodns/processor/arpa.py (hunk 1 above)
Comment (wjgauthier) [reply octodns/processor/arpa.py]: Yes, I think you are right. This will only keep things unique if they have the same priority. I have moved it to populate instead.

File: octodns/processor/arpa.py [hunk 2]
```diff
@@ -56,7 +59,11 @@ def populate(self, zone, target=False, lenient=False):
         for arpa, fqdns in self._records.items():
             if arpa.endswith(f'.{zone_name}'):
                 name = arpa[:-n]
+                fqdns = list(set(fqdns))
```
Comment (ross) [inline octodns/processor/arpa.py:62]: I think this will still have the problem of including the priority in the uniqueness. I believe the uniq-ifying would need to happen after line 64 where the `d[1]` bit is extracted, but that would potentially lose the sort order so this one is kind of interesting. 

May be easiest/cleanest to switch lines 63-65 to a more explicit setup like:

```python
seen = set()
ordered = sorted(fqdns)
fqdns = []
for _, fqdn in ordered:
    if fqdn in seen:
        continue
    fqdns.append(fqdn)
    seen.add(fqdn)
    if len(seen) >= self.max_auto_arpa:
        break
```

The trickiness of this definitely points towards the need for some specific testing around the behavior with duplicates fqdns & priorities, fqdns w/different priorities, etc. 

Might make sense to pull the logic out into a helper function:

```python
# Note: this takes a list of (priority, fqdn) tuples and returns the ordered and uniqified list of fqdns.
fqdns = self._order_and_unique_fqdns(fqdns, self.max_auto_arpa)
```

It would then be easy to call this method directly on an `AutoArpa` instance with all the various permutations without having to deal with creating records and all of the unrelated scaffolding.

**Review by ross — COMMENTED** [review]
Comment/suggestion inline.

File: octodns/processor/arpa.py (hunk 2 above)
Comment (wjgauthier) [reply octodns/processor/arpa.py]: Great, thank you. I have added the suggestion.

**Review by ross — APPROVED** [review]

**Comment by ross** [comment]
googlecloud failure is unrelated, just dep problems in all of their modules.
